File: Theatre Plays - Lab 10/webapp/src/app/directors/shared/director.service.ts
import {Injectable} from "@angular/core";
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {Director} from "./director.model";

@Injectable()
export class DirectorService {

  private directorsUrl = 'http://localhost:8080/api/directors';

  constructor(private httpClient: HttpClient) {
  }

  getDirectors(): Observable<Director[]> {
    return this.httpClient
      .get<Array<Director>>(this.directorsUrl);
  }

  addDirector(director: Director): Observable<Director> {
    return this.httpClient
      .post<Director>(this.directorsUrl, director);
  }
}

File: Theatre Plays - Lab 10/catalog-core/src/main/java/ro/ubb/catalog/core/service/DirectorService.java
package ro.ubb.catalog.core.service;

import ro.ubb.catalog.core.model.Actor;
import ro.ubb.catalog.core.model.Director;

import java.util.List;

public interface DirectorService {
  Director addDirector(Director director);
  List<Director> getAllDirectors();
}
